app-android/src/main/java/com/ustadmobile/port/android/view/ContentEntryList2Fragment.kt

```diff
@@ -139,8 +139,16 @@ class ContentEntryList2Fragment : UstadListViewFragment<ContentEntry, ContentEnt
 
     override fun showContentEntryAddOptions(parentEntryUid: Long) {
         val entryAddOption = ContentEntryAddOptionsBottomSheetFragment()
-        entryAddOption.arguments = UMAndroidUtil.mapToBundle(mapOf(UstadView.ARG_PARENT_ENTRY_UID
-                to parentEntryUid.toString()))
+        val args = mutableMapOf(
+                UstadView.ARG_PARENT_ENTRY_UID to parentEntryUid.toString())
+
+        // add selectedItems to args if in stack
+        if(findNavController().previousBackStackEntry?.savedStateHandle?.contains(KEY_SELECTED_ITEMS) == true){
```

Comment:
Don't look back - pass the arguments forward. This can be done using the multiplatform navigate for results.